src/util/bip32.rs

```diff
@@ -337,33 +338,36 @@ impl ExtendedPubKey {
     }
 }
 
-impl ToBase58 for ExtendedPrivKey {
-    fn base58_layout(&self) -> Vec<u8> { 
-        let mut ret = Vec::with_capacity(78);
-        ret.extend(match self.network {
-            Network::Bitcoin => [0x04, 0x88, 0xAD, 0xE4],
-            Network::Testnet => [0x04, 0x35, 0x83, 0x94]
-        }.iter().cloned());
-        ret.push(self.depth as u8);
-        ret.extend(self.parent_fingerprint[..].iter().cloned());
+impl ToString for ExtendedPrivKey {
+    fn to_string(&self) -> String {
+        let mut ret = [0; 78];
+        ret[0..4].copy_from_slice(match self.network {
+            Network::Bitcoin => &[0x04, 0x88, 0xAD, 0xE4][..],
+            Network::Testnet => &[0x04, 0x35, 0x83, 0x94][..],
```

Comment:
I get compile errors here on branch bech32:

 --> src/util/bip32.rs:345:61
    |
345 |             Network::Bitcoin => &[0x04, 0x88, 0xAD, 0xE4][..],
    |                                  ------------------------   ^ temporary value dropped here while still borrowed
    |                                  |
    |                                  temporary value created here
346 |             Network::Testnet => &[0x04, 0x35, 0x83, 0x94][..],
347 |         });
    |          - temporary value needs to live until here

error[E0597]: borrowed value does not live long enough
   --> src/util/bip32.rs:346:61
    |
346 |             Network::Testnet => &[0x04, 0x35, 0x83, 0x94][..],
    |                                  ------------------------   ^ temporary value dropped here while still borrowed
    |                                  |
    |                                  temporary value created here
347 |         });
    |          - temporary value needs to live until here

error[E0597]: borrowed value does not live long enough
   --> src/util/bip32.rs:403:63
    |
403 |             Network::Bitcoin => &[0x04u8, 0x88, 0xB2, 0x1E][..],
    |                                  --------------------------   ^ temporary value dropped here while still borrowed
    |                                  |
    |                                  temporary value created here
404 |             Network::Testnet => &[0x04u8, 0x35, 0x87, 0xCF][..],
405 |         });
    |          - temporary value needs to live until here

error[E0597]: borrowed value does not live long enough
   --> src/util/bip32.rs:404:63
    |
404 |             Network::Testnet => &[0x04u8, 0x35, 0x87, 0xCF][..],
    |                                  --------------------------   ^ temporary value dropped here while still borrowed
    |                                  |
    |                                  temporary value created here
405 |         });
    |          - temporary value needs to live until here

error: aborting due to 4 previous errors

Reply:
running rustc 1.20 on osx

Reply:
Thanks, works on rustc 1.24, I'll test with 1.20.

Reply:
Can reproduce. Will fix.

Reply:
Pushed fix which works back to 1.16 at least (though the bech32 dep does not because it uses implicit struct names, something clippy recommends even though it breaks backward compat).